clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java

```diff
@@ -733,17 +733,26 @@ protected void handleRoleCommand() {
     String userName = line.getOptionValue(GravitinoOptions.LOGIN);
     FullName name = new FullName(line);
     String metalake = name.getMetalakeName();
-    String role = line.getOptionValue(GravitinoOptions.ROLE);
     String[] privileges = line.getOptionValues(GravitinoOptions.PRIVILEGE);
 
     Command.setAuthenticationMode(auth, userName);
 
+    String[] roles = line.getOptionValues(GravitinoOptions.ROLE);
+    if (roles == null && !CommandActions.LIST.equals(command)) {
+      System.err.println(ErrorMessages.MISSING_ROLE);
+      Main.exit(-1);
+    }
```

Comment:
I don;t this code will never be run, if an option exists it can never be null as the CLI library catches this error.

Reply:
In the same spirit, I'm curious whether/how we can generalize this checking of duplicated option which is supposed to be specified only once.
This is not specific to the `--role` option, and we are not supposed to handle them for all actions separately.

Reply:
The CLI library handles that. Generally, things can only be specified once unless you say having multiple values is OK.

Reply:
> I don't think this code will ever be run. If an option exists, it can never be null, as the CLI library catches this error.

IHMO, A friendly hint should be provided if the user forgets to pass the -role option. like `gcli role details -m demo_metalake`,

Reply:
I'm not in front of my work computer, but please double-check that the CLI doesn't give an error if there is no role option. There is no need to try and do things that the CLI already checks.

Reply:
Check both `gcli role details -m demo_metalake` and `gcli role details -m demo_metalake --role`.

Reply:
![image](https://github.com/user-attachments/assets/f975a316-6517-4913-883b-67729f5ae6b0)
Compiled result from the master branch. If the client fails to provide the `--role` option, an Unfriendly information will be shown.
If we add checking logic, the information be displayed as follows:
```bash
gcli role details -m demo_metalake
Missing --role option.
```

Reply:
that is expected to be fixed by #5832

Reply:
Main Branch result:
```bash
gcli role details -m demo_metalake
# invalid string to encode: null

gcli role details -m demo_metalake --role
# Error parsing command line: Missing argument for option: r
```

Reply:
Leave your check-in for now, we can improve it later.